deploy/server.conf

```diff
@@ -0,0 +1,100 @@
+# Some elements here taken from https://github.com/nystudio107/nginx-craft/blob/master/sites-available/somedomain.com.conf
+
+# Bots to ban via user agent
+map $http_user_agent $limit_bots {
+     default 0;
+     ~*(AhrefsBot|Baiduspider|PaperLiBot) 1;
+}
+
+server {
+    listen 80;
+    listen [::]:80;
+    root "/var/www/craft/web";
+    index index.html index.htm index.php;
+    charset utf-8;
+
+    # Enable serving of static gzip files as per:
+    # http://nginx.org/en/docs/http/ngx_http_gzip_static_module.html
+    gzip_static on;
+
+    # Enable server-side includes as per:
+    # http://nginx.org/en/docs/http/ngx_http_ssi_module.html
+    ssi on;
+
+    # Disable limits on the maximum allowed size of the client request body
+    client_max_body_size 0;
+
+    # Ban certain bots from crawling the site
+    if ($limit_bots = 1) {
+        return 403;
+    }
+
+    # 404 error handler
+    error_page 404 /index.php?$query_string;
+
+    # 301 Redirect URLs with trailing /'s as per
+    # https://webmasters.googleblog.com/2010/04/to-slash-or-not-to-slash.html
+    rewrite ^/(.*)/$ /$1 permanent;
+
+    # Change // -> / for all URLs, so it works for our php location block, too
+    merge_slashes off;
+    rewrite (.*)//+(.*) $1/$2 permanent;
+
+    # For WordPress bots/users
+    location ~ ^/(wp-login|wp-admin|wp-config|wp-content|wp-includes|xmlrpc) {
+        return 301 https://wordpress.com/wp-login.php;
+    }
+
+    # Handle Do Not Track as per https://www.eff.org/dnt-policy
+    location /.well-known/dnt-policy.txt {
+        try_files /dnt-policy.txt /index.php?p=/dnt-policy.txt;
+    }
+
+    # Don't send the nginx version number in error pages and Server header
+    server_tokens off;
+
+    # Root directory location handler
+    location / {
+        try_files $uri/index.html $uri $uri/ /index.php?$query_string;
+    }
+
+    # Craft-specific location handlers to ensure AdminCP requests route through index.php
+    # If you change your `cpTrigger`, change it here as well
+    location ^~ /admin {
+        try_files $uri $uri/ /index.php?$query_string;
+    }
+    location ^~ /cpresources {
+        try_files $uri $uri/ /index.php?$query_string;
+    }
+
+    # Add rate limiting for potentially-expensive operations (eg. API calls)
+    location ~ ^/api {
+        limit_req zone=api_routes burst=20 nodelay;
```

Comment:
Get the intent here but rather than rate-limited I'd look at using the cache functionality here https://www.nginx.com/blog/nginx-caching-guide/ although we likely need to use the `fastcgi_` prefixed versions rather than `proxy_` used in that document

Reply:
Could you give this another look? Gave it a tweak and happier with this now.